nucleus/deploy/model_endpoint.py

```diff
@@ -1,12 +1,41 @@
 import concurrent.futures
 from collections import Counter
-from typing import Dict, Optional, Sequence
+from typing import Any, Dict, Optional, Sequence
 
 TASK_PENDING_STATE = "PENDING"
 TASK_SUCCESS_STATE = "SUCCESS"
 TASK_FAILURE_STATE = "FAILURE"
 
 
+class EndpointRequest:
+    """
+    Represents a single request to either a SyncModelEndpoint or AsyncModelEndpoint.
+    Parameters:
+        url: A url to some file that can be read in to a ModelBundle's predict function. Can be an image, raw text, etc.
+        args: A Dictionary with arguments to a ModelBundle's predict function. If the predict function has signature
+            predict_fn(foo, bar), then the keys in the dictionary should be 'foo' and 'bar'. Values must be native Python
+            objects.
+        return_pickled: Whether the output should be a pickled python object, or directly returned serialized json
+        request_id: A user-specifiable id for requests. Should be unique.
+    """
+
+    def __init__(
+        self,
+        url: Optional[str] = None,
+        args: Optional[Dict] = None,
```

Comment:
Wonder if we should have these two things be a `Union` (we'd still have to do the XOR check because `Union` is really just a hint, from a type annotation perspective).

Reply:
I sorta am leaning away from putting the two into a `Union`. We'd have to keep track of whether the data represents a url or args (which we technically could do by looking at their types, but this feels wrong), and keeping them separate seems like the easiest way to do this. The stuff stored at url/args gets put into separate fields in the eventual http request's json anyway, so I think it's simpler to keep things separate.

Reply:
Yup that makes sense.